src/context.rs

```diff
@@ -93,13 +93,17 @@ impl RealTimeDataset {
 
 #[derive(Debug, Clone)]
 pub struct Period {
-    pub begin: NaiveDate,
-    pub end: NaiveDate,
+    pub begin: Option<NaiveDate>,
```

Comment:
why do you pass an option here ? it would be easier to init it at each loading once at `today` .

the current implementation can lead to very strange problem if we generate the dataset around midnigh and the day change while building the timetable.

Reply:
what do call a loading? when loading the dataset every 24h? If so, I don’t see a difference with the current approach.

If you mean in the main.rs, then we have the current bug that 48h later we won’t have any data as the period is wrong.

The only reason to define a begin is to be able to have a specific test for the tests.

Reply:
my point is that in this implementation we compute the `local::Today` at every stoptime, so we have a theorically shifting filter (even if for the moment the timetable creation is quite fast, I think this can lead to quite strange bug).

I this you can just set the begin at the Period creation [at init](https://github.com/etalab/transpo-rt/blob/aa201cda8cf23a3d5b141d4b6a06be71e907bace/src/main.rs#L47) and [at reloading](https://github.com/etalab/transpo-rt/blob/master/src/actors/update_actors.rs#L22)

Reply:
ooooh ! got it :)

Reply:
better like this?

Reply:
yup 😄 

I'm sorry for being picky, but with the current implementation, why do we need an Option ?
by always having a `begin` (and creating it at `today()` while reloading) we can track it in the `/status` (so the `begin` is not useful only for the tests)

Reply:
I think I found out what bugs me:
* the start date and horizon are configurations, maybe they should be defined there?
* the dataset doesn’t really need to know what it is, just the updater, and that would simplify throwing the data all around (but we would need to change /status)

Reply:
We good, now ? :D

Reply:
great, thanks!